stdlib/http/src/main/java/org/ballerinalang/net/http/HttpUtil.java

```diff
@@ -1534,6 +1535,13 @@ public static ListenerConfiguration getListenerConfig(long port, Struct endpoint
 
         listenerConfiguration.setServerHeader(getServerName());
 
+        //check whether http configuration or grpc configuration
+        if (endpointConfig.getFields().length > 7) {
```

Comment:
Shall we use following logic rather checking for no of fields?

```
        if (Arrays.asList(endpointConfig.getFields()).contains(SERVER_NAME)) {
           Value serverName = endpointConfig.getRefField(SERVER_NAME);
            listenerConfiguration.setServerHeader(serverName != null ? serverName.getStringValue() : getServerName());
        } else {
            listenerConfiguration.setServerHeader(getServerName());
        }
```

And then we can remove line no 1536 too